File: app/src/main/java/com/seventhourdev/quickcalendar/models/Event.java
package com.seventhourdev.quickcalendar.models;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by david on 8/5/2016.
 */
public class Event implements Parcelable {
    public String start, end;
    protected Event(Parcel in) {
    }

    public static final Creator<Event> CREATOR = new Creator<Event>() {
        @Override
        public Event createFromParcel(Parcel in) {
            return new Event(in);
        }

        @Override
        public Event[] newArray(int size) {
            return new Event[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
    }
}

File: app/src/main/java/com/seventhourdev/quickcalendar/models/Event.kt
package com.seventhourdev.quickcalendar.models

import io.realm.RealmObject
import android.os.Parcel
import android.os.Parcelable
import io.realm.annotations.Ignore
import io.realm.annotations.RealmClass
import java.security.Timestamp
import java.util.*

/**
 * Created by david on 8/5/2016.
 */
open class Event(open var startLong: Long = GregorianCalendar().timeInMillis,
                 open var endLong: Long=GregorianCalendar().timeInMillis,
                 @Ignore open var start: GregorianCalendar = GregorianCalendar(),
                 @Ignore open var end: GregorianCalendar = GregorianCalendar()
                 ) : RealmObject(){
    public fun upDateTimes(newStart:GregorianCalendar, newEnd:GregorianCalendar){
        start=newStart
        end=newEnd
        startLong=newStart.timeInMillis
        endLong=newEnd.timeInMillis
    }
}
